mhkit/mooring/general.py

```diff
@@ -0,0 +1,54 @@
+import xarray as xr
+
+
+def laylength(ds, depth, tolerance=0.25):
+    """Calculate the laylength of a mooring line over time.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Dataset containing x,y,z nodes (ie Node1px, Node1py, Node1pz)
+    depth : float
+        Depth of seabed (m)
+    tolerance : float, optional
+        Tolerance to detect first lift point from seabed, by default 0.25 meters
+
+    Returns
+    -------
+    xr.Dataset
+        Array containing the laylength at each time step.
+
+    Raises
+    ------
+    ValueError
+        Checks for mininum number of nodes necessary to calculate laylength.
```

Comment:
This is a great addition. Based on the change below I think would be updated to:

```
Raises
------
ValueError
    Checks for minimum number of nodes necessary to calculate laylength.
TypeError
    Checks for correct input types for ds, depth, and tolerance.
    ```